src/live_restore/live_restore_fs.c

```diff
@@ -729,36 +819,55 @@ __live_restore_fh_read(
 int
 __wti_live_restore_fs_fill_holes(WT_FILE_HANDLE *fh, WT_SESSION *wt_session)
 {
+    WT_DECL_RET;
 /*
  * Holes can be large, potentially the size of an entire file. When we find a large hole we'll read
- * it in 4KB chunks.
+ * it in 4KB chunks. This function will take the extent list writelock. The caller should *not* hold
+ * the lock when calling.
  */
 #define WT_LIVE_RESTORE_READ_SIZE ((size_t)(4 * WT_KILOBYTE))
     WT_LIVE_RESTORE_FILE_HANDLE *lr_fh;
     WT_LIVE_RESTORE_HOLE_NODE *hole;
 
     char buf[WT_LIVE_RESTORE_READ_SIZE];
     lr_fh = (WT_LIVE_RESTORE_FILE_HANDLE *)fh;
+    WT_SESSION_IMPL *session = (WT_SESSION_IMPL *)wt_session;
 
-    while ((hole = lr_fh->destination.hole_list_head) != NULL) {
-        __wt_verbose_debug3((WT_SESSION_IMPL *)wt_session, WT_VERB_FILEOPS,
+    while (true) {
+        __wt_writelock(session, &lr_fh->ext_lock);
```

Comment:
This should be able to use the `WT_WITH...` macro with a bit of refactoring?

Reply:
12390ade1